b2sdk/encryption/setting.py

```diff
@@ -54,29 +97,78 @@ def __eq__(self, other):
         return self.mode == other.mode and self.algorithm == other.algorithm and self.key == other.key
 
     def as_value_dict(self):
+        """
+        Dump EncryptionSetting as dict for serializing a to json for requests.
+        """
+        if self.key and self.key.secret is None:
+            raise ValueError('cannot use an unknown key in transmission')
```

Comment:
as_value_dict() doesn't know who is calling it and why, that's why
EncryptionSetting.add_to_download_headers
EncryptionSetting.add_to_upload_headers
EncryptionSetting._add_sse_b2_headers
EncryptionSetting._add_sse_c_headers
should make sure they are doing something meaningful. If anyone is calling as_value_dict() to put something for transmitting a request over to the server, it should probably be a callback to EncryptionSettingHelper or something which should contain the logic on manipulating headers, POST data etc for using encryption

Reply:
okay, so this relates to the comment concerning `serialize_to_json_for_request`. What do you mean by `should make sure they are doing something meaningful` ? Does refactoring `repr_as_dict,as_value_dict` -> `as_dict,serialize_to_json_for_request` make it okay?

Reply:
sort of. It's a good start. As we discussed there should also be a delegate for attaching encryption settings object to a given AbstractRawApi implementation